### PR TITLE
Gh 1105 extra default parquet serialisers

### PR DESCRIPTION
**Pull request by m55624**

### CODE REVIEW

File: store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/serialisation/impl/ArrayListStringParquetSerialiser.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        }
+        return new Object[]{null};
+    }
+
+    @Override
+    public ArrayList<String> deserialise(final Object[] objects) throws SerialisationException {
+
+        if (objects.length == 1) {
+            if (objects[0] instanceof String[]) {
+                final ArrayList<String> arrayList = new ArrayList<>();
```
Comment (ac74475) [inline store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/serialisation/impl/ArrayListStringParquetSerialiser.java]: Given you know the size of the ArrayList, you could construct the ArrayList to the desired size.

File: store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/serialisation/impl/HashSetStringParquetSerialiser.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            return new Object[]{objects};
+        }
+        return new Object[]{null};
+    }
+
+    @Override
+    public HashSet<String> deserialise(final Object[] objects) throws SerialisationException {
+        if (objects.length == 1) {
+            if (objects[0] instanceof String[]) {
+                final HashSet<String> hashSet = new HashSet<>();
```
Comment (ac74475) [inline store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/serialisation/impl/HashSetStringParquetSerialiser.java]: Same as the previous comment

**Comment by m55624** [comment]
Spoken to @p013570 who has said the canHandle should contain the equals method instead of isAssignableFrom, I have updated this for the TreeSetSerialiser too.

File: store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/serialisation/impl/ArrayListStringParquetSerialiser.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    @Override
+    public ArrayList<String> deserialise(final Object[] objects) throws SerialisationException {
+
+        if (objects.length == 1) {
+            if (objects[0] instanceof String[]) {
+                final String[] objectsToDeserialise = (String[]) objects[0];
+                final ArrayList<String> arrayList = new ArrayList<>(objectsToDeserialise.length);
+                arrayList.addAll(Arrays.asList(objectsToDeserialise));
+                return arrayList;
```
Comment (p013570) [inline store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/serialisation/impl/ArrayListStringParquetSerialiser.java]: just use:
return Lists.newArrayList(objectsToDeserialise);

File: store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/serialisation/impl/HashSetStringParquetSerialiser.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        }
+        return new Object[]{null};
+    }
+
+    @Override
+    public HashSet<String> deserialise(final Object[] objects) throws SerialisationException {
+        if (objects.length == 1) {
+            if (objects[0] instanceof String[]) {
+                final String[] objectsToDeserialise = (String[]) objects[0];
+                final HashSet<String> hashSet = new HashSet<>(objectsToDeserialise.length);
```
Comment (p013570) [inline store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/serialisation/impl/HashSetStringParquetSerialiser.java]: just use:

return Sets.newHashSet(objectsToDeserialise);

File: store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/serialisation/impl/TypeSubTypeValueParquetSerialiser.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        return new Object[]{null, null, null};
+    }
+
+    @Override
+    public TypeSubTypeValue deserialise(final Object[] objects) throws SerialisationException {
+        if (objects.length == 3) {
+            if (objects[0] instanceof String
+                    && objects[1] instanceof String
+                    && objects[2] instanceof String) {
+                return new TypeSubTypeValue((String) objects[0], (String) objects[1], (String) objects[2]);
```
Comment (p013570) [inline store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/serialisation/impl/TypeSubTypeValueParquetSerialiser.java]: Could you remove the instanceof checks and just do:

return new TypeSubTypeValue(objects[0].toString(), objects[1].toString(), objects[2].toString());

**Pull request merged**